### ISSUE TITLE
getUniformLocation should not require [0] suffix for array uniforms

### ISSUE THREAD

**Issue by leosingleton** [stackgl/headless-gl#170]
When porting some existing code to run under headless-gl, I was receiving a null return code from getUniformLocation() on array uniforms. The root cause was not adding a [0] suffix:

`gl.getUniformLocation(program, "arrayUniform")` <-- Returns null
`gl.getUniformLocation(program, "arrayUniform[0]")` <-- Works as expected

According to the docs, the [0] suffix should be optional: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/getUniformLocation
>Additionally, for uniforms declared as arrays, the following names are also valid:
>
> The uniform name without the [0] suffix. E.g. the location returned for arrayUniform is equivalent to the one for arrayUniform[0].

**Comment by dhritzkiv** [comment]
Thanks for bringing this to our attention. This may be the result of us using an older version of angle (our gl implementation), which may not implement the standard faithfully.

However, it may be possible to implement this on the JS side, here:
https://github.com/stackgl/headless-gl/blob/adbf6ab3ac7cf652dac6dfc24b78ef6af2cccfba/src/javascript/webgl-rendering-context.js#L2574

**Comment by leosingleton** [comment]
Yes, when I was stepping through the same function, I'm pretty sure the issue is in the JS side. The programs._uniforms[] array had all of the uniforms listed--the array one just didn't match in the for loop without the [0] suffix. I was hoping to help fix it on the headless-gl side, but ran into issues compiling angle locally, so it was just easier to work around it.

And thanks, it's a great project! Other than this minor bug I had to work around, existing WebGL code just runs under Node!

**Comment by wcandillon** [comment]
@akira-cn thank you for the workaround, it works great! Hope this gets published in the library. I love this lib by the way!
